### PR TITLE
[PVR] Separate GUI from PVR core: remove CFileItem usage from core: step 1: directory listings.

### PR DESCRIPTION
**Pull request by ksooo**
Backport of #16170.

@Jalle19 Please aprove.

### CODE REVIEW

**Comment by ksooo** [comment]
jenkins build this please

**Review by Jalle19 — APPROVED** [review]
Seems like quite a big change to backport, but it's your call

File: xbmc/pvr/PVRGUIDirectory.cpp [hunk 1]
```diff
@@ -0,0 +1,504 @@
...
+
+bool CPVRGUIDirectory::HasDeletedRadioRecordings()
+{
+  return CServiceBroker::GetPVRManager().IsStarted() &&
+         CServiceBroker::GetPVRManager().Recordings()->HasDeletedRadioRecordings();
+}
+
+namespace
+{
+  std::string TrimSlashes(const std::string& strOrig)
```
Comment (Rechi) [inline xbmc/pvr/PVRGUIDirectory.cpp]: https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#32-indentation
> Do not indent namespaces to simplify nesting them and wrapping .cpp files in a namespace.

File: xbmc/pvr/PVRGUIDirectory.cpp [hunk 2]
```diff
@@ -0,0 +1,504 @@
...
+    FilterDirectory(results);
+    return true;
+  }
+
+  return false;
+}
+
+namespace
+{
+  bool GetTimersRootDirectory(const CPVRTimersPath& path,
```
Comment (Rechi) [inline xbmc/pvr/PVRGUIDirectory.cpp]: https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#32-indentation
> Do not indent namespaces to simplify nesting them and wrapping .cpp files in a namespace.

File: xbmc/pvr/PVRGUIDirectory.h [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+  /*!
+   * @brief Check if any deleted radio recordings are existing.
+   * @return True if deleted radio recordings exists, false otherwise.
+   */
+  static bool HasDeletedRadioRecordings();
+
+private:
+
+  bool FilterDirectory(CFileItemList &results) const;
```
Comment (Rechi) [inline xbmc/pvr/PVRGUIDirectory.h]: left align `&`

File: xbmc/pvr/PVRGUIDirectory.cpp [hunk 4]
```diff
@@ -0,0 +1,504 @@
...
+    bool bHideDisabled = CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(CSettings::SETTING_PVRTIMERS_HIDEDISABLEDTIMERS);
+
+    std::shared_ptr<CFileItem> item;
+
+    for (const auto& timer : timers)
+    {
+      if ((timer->m_bIsRadio == bRadio) &&
+          (timer->m_iParentClientIndex != PVR_TIMER_NO_PARENT) &&
+          (timer->m_iClientId == iClientId) &&
+          (timer->m_iParentClientIndex == iParentId) &&
```
Comment (Rechi) [inline xbmc/pvr/PVRGUIDirectory.cpp]: > `warning: comparison between signed and unsigned integer expressions [-Wsign-compare]`

Comment (ksooo) [reply xbmc/pvr/PVRGUIDirectory.cpp]: Yeah, that's why the jenkins build fails. I will fix that, ofc.

**Comment by ksooo** [comment]
@Rechi requested changes are in. Will open a respective PR for master, where same improvements should be done.

**Comment by Rechi** [comment]
wrong revision was build
jenkins build this please

**Pull request merged**